packages/core/src/index.ts

```diff
@@ -30,6 +30,10 @@ class ImageService extends Service {
     return this.caller.collect('booru', () => delete this.sources[id])
   }
 
+  check() {
```

Comment:
The function here is not good. Either change the name to something like `hasSource`, or remove it and check `this.counter in the action part.

Reply:
change it to `hasSource`, I think it will be used in feature